public/app/plugins/datasource/opentsdb/datasource.js

```diff
@@ -50,6 +53,9 @@ function (angular, _, dateMath) {
           if (index === -1) {
             index = 0;
           }
+
+          saveTagKeys(metricData);
```

Comment:
please use this._saveTagKeys  instead, and bind enclosing function to this (will let you remove the saveTagKeys variable at the top)